python/ray/train/torch.py

```diff
@@ -74,6 +87,37 @@ def prepare_model(
         if move_to_device:
             logger.info(f"Moving model to device: {device}")
             model = model.to(device)
+
+        def wrap_forward(forward):
+            @functools.wraps(forward)
+            def wrapper(*args, **kwargs):
+                with autocast():
+                    outputs = forward(*args, **kwargs)
+                assert isinstance(outputs, torch.Tensor)
+                return outputs.float()
+
+            return wrapper
+
+        def model_get_state(self):
```

Comment:
Had to invoke some supernatural powers to implement this hack. Once we've figured out checkpointing for Ray ML, we should refactor this hack.